src/components/ListItem.jsx

```diff
@@ -57,22 +97,100 @@ export function ListItem({ name, listToken, itemId, data, urgency }) {
 		}
 	};
 
+	const computeDate = (secs) => {
+		const output = new Date(secs * 1000).toString().split(' ');
+		return `${output[1]} ${output[2]}, ${output[3]}`;
+	};
+
+	const lastPurchase = data.dateLastPurchased
+		? computeDate(data.dateLastPurchased.seconds)
```

Comment:
Instead of returning 'N/A', you can use optional chaining operator to return undefined if dateLastPurchased is undefined 
```suggestion
		? computeDate(data.dateLastPurchased?.seconds)
```

Reply:
Seeing "undefined" on the UI may not make sense to the user, they might interpret it as an error/bug in the application.